graphqlmetrics/core/telemetry_test.go

```diff
@@ -0,0 +1,85 @@
+package core
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"testing"
+	"time"
+
+	"connectrpc.com/connect"
+	"github.com/wundergraph/cosmo/graphqlmetrics/pkg/telemetry"
+	"go.uber.org/zap"
+
+	v1 "github.com/wundergraph/cosmo/graphqlmetrics/gen/proto/wg/cosmo/graphqlmetrics/v1"
+)
+
+type mockMS struct{}
+
+func (ms *mockMS) PublishGraphQLMetrics(context.Context, *connect.Request[v1.PublishGraphQLRequestMetricsRequest]) (*connect.Response[v1.PublishOperationCoverageReportResponse], error) {
+	return nil, nil
+}
+
+func TestPrometheusExport(t *testing.T) {
+	ms := &mockMS{}
+	type test struct {
+		svr     *Server
+		name    string
+		wantErr bool
+	}
+	tests := []test{
+		{
+			name: "with-prometheus-server-enabled",
+			svr: NewServer(ms, WithMetrics(
+				telemetry.NewTelemetryConfig(
+					telemetry.PrometheusConfig{
+						Enabled:    true,
+						ListenAddr: "127.0.0.1:8089",
+						Path:       "/metrics",
+					},
+				),
+			)),
+			wantErr: false,
+		},
+		{
+			name: "without-prometheus-server-enabled",
+			svr: NewServer(ms, WithMetrics(
+				telemetry.NewTelemetryConfig(
+					telemetry.PrometheusConfig{
+						Enabled:    false,
+						ListenAddr: "127.0.0.1:8089",
+						Path:       "/metrics",
+					},
+				),
+			)),
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		// as the server setup is all encapsulated within main,
+		// at least check, if the new code is covered and if the
+		// handlers for the prometheus server are registered
+		go func() {
+			if err := tt.svr.StartPrometheusServer(); err != nil {
```

Comment:
Please add an integration test that actually test the server implementation. Additionally, we want to validate in the tests that proper metric datapoints has been exposed.

```go
    // 1. Setup the server
	svr := core.NewServer(
		ms,
	   core.WithMetrics(metricsConfig),
		core.WithListenAddr(cfg.ListenAddr),
	)

// 2. Make a request to the buf service
// 3. Validate metrics on the prometheus endpoint
```

Reply:
As far as I see, in graphqlmetrics, there are only Integration Tests.

Is this ok to only have those here e.g. mocking the MetricsService seemed odd to me in the first place already to get some coverage here.